great_expectations/profiler/domain_builder/util.py

```diff
@@ -0,0 +1,100 @@
+from typing import Any, Dict, List, cast
+
+import great_expectations.exceptions as ge_exceptions
+from great_expectations.core.domain_types import SemanticDomainTypes
+from great_expectations.validator.validation_graph import MetricConfiguration
+from great_expectations.validator.validator import Validator
+
+
+def translate_table_column_type_to_semantic_domain_type(
+    validator: Validator, column_name: str
+) -> SemanticDomainTypes:
+    # Note: As of Python 3.8, specifying argument type in Lambda functions is not supported by Lambda syntax.
+    column_types_dict_list: List[Dict[str, Any]] = list(
+        filter(
+            lambda column_type_dict: column_name in column_type_dict,
+            validator.get_metric(
+                metric=MetricConfiguration(
+                    metric_name="table.column_types",
+                    metric_domain_kwargs={},
+                    metric_value_kwargs={
+                        "include_nested": True,
+                    },
+                    metric_dependencies=None,
+                )
+            ),
+        )
+    )
+    if len(column_types_dict_list) != 1:
+        raise ge_exceptions.ProfilerExecutionError(
+            message="A unique column could not be found while obtaining semantic type information."
+        )
+
+    column_type: str = cast(str, column_types_dict_list[0][column_name]).upper()
+
+    semantic_column_type: SemanticDomainTypes
+    if column_type in [
+        "UNIQUEIDENTIFIER",
+    ]:
+        semantic_column_type = SemanticDomainTypes.IDENTITY
+    elif column_type in [
+        "NUMERIC",
+        "BINARY",
+        "VARBINARY",
+        "INT",
+        "INT64",
+        "INTEGER",
+        "TINYINT",
+        "SMALLINT",
+        "BIGINT",
+        "DECIMAL",
+        "DOUBLE",
+        "DOUBLE_PRECISION",
+        "FLOAT",
+        "FLOAT64",
+        "REAL",
```

Comment:
I wonder if for this (and the below types) we can instead refer to the types in ProfilerTypeMapping from `profile/base.py`. I believe James wanted to reuse those, and it would be fine to add to them if they're not exhaustive enough. If the structure is too different, then we can recreate as you're doing here.

Reply:
Great point -- in next commit.